BeagleBone_black/src/navigation/navigation.cpp

```diff
@@ -78,6 +78,18 @@ NavigationType Navigation::getEmergencyBrakingDistance() const
   return velocity_[0]*velocity_[0] / kEmergencyDeceleration;
 }
 
+NavigationType Navigation::getBrakingDistance() const
+{
+  // A polynomial fit for the braking distance at a specific (normalised) velocity
+  NavigationType norm_v = (getVelocity() - 45.5628) / 21.9511;
+  NavigationType braking_distance = 2.0;
+  for (unsigned int i = 0; i < coefficients.size(); ++i) {
+    braking_distance += coefficients[i] * pow(norm_v, i);
```

Comment:
Use of `pow` here is quite inefficient. You can instead do an update like `norm_v *= norm_v`  every iteration. You'd just need to handle the constant term separately.